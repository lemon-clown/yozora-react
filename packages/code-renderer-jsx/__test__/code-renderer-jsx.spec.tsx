import { mount } from 'enzyme'
import React from 'react'
import CodeRendererJsx from '../src'

describe('basic rendering case', () => {
  const errorLogger = jest
    .spyOn(global.console, 'error')
    .mockImplementation((...args) => {
      throw new Error(args.join(' '))
    })

  afterAll(() => {
    errorLogger.mockRestore()
  })

  it('snapshot', () => {
    function Wrapper(props: {
      code: string
      inline: boolean
    }): React.ReactElement {
      const [error, setError] = React.useState<any>(null)
      return (
        <div>
          <CodeRendererJsx
            code={props.code}
            inline={props.inline}
            onError={setError}
          />
          <pre>{error}</pre>
        </div>
      )
    }

    const inlineCode = `
      (
        <div>
          <span>Hello, world</span>
        </div>
      )
    `
    expect(mount(<Wrapper code={inlineCode} inline={true} />)).toMatchSnapshot(
      'inline code',
    )

    const blockCode = `
      const Container = styled.div\`
        background: hsl(0deg, 10%, 90%);
      \`

      render(
        <Container>
          <span style={{ color: 'orange' }}>Hello, world</span>
        </Container>
      )
    `
    expect(mount(<Wrapper code={blockCode} inline={false} />)).toMatchSnapshot(
      'block code',
    )

    const badCode = `
      (<div>hello</div>)
    `
    expect(mount(<Wrapper code={badCode} inline={false} />)).toMatchSnapshot(
      'bad code',
    )
  })
})
