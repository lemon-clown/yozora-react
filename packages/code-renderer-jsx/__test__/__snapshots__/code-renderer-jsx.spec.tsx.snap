// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic rendering case snapshot: bad code 1`] = `
<Wrapper
  code="
      (<div>hello</div>)
    "
  inline={false}
>
  <div>
    <YozoraCodeRendererJsx
      code="
      (<div>hello</div>)
    "
      inline={false}
      onError={[Function]}
      scope={Object {}}
    />
    <pre>
      SyntaxError: No-Inline evaluations must call \`render\`.
    </pre>
  </div>
</Wrapper>
`;

exports[`basic rendering case snapshot: block code 1`] = `
<Wrapper
  code="
      const Container = styled.div\`
        background: hsl(0deg, 10%, 90%);
      \`

      render(
        <Container>
          <span style={{ color: 'orange' }}>Hello, world</span>
        </Container>
      )
    "
  inline={false}
>
  <div>
    <YozoraCodeRendererJsx
      code="
      const Container = styled.div\`
        background: hsl(0deg, 10%, 90%);
      \`

      render(
        <Container>
          <span style={{ color: 'orange' }}>Hello, world</span>
        </Container>
      )
    "
      inline={false}
      onError={[Function]}
      scope={Object {}}
    />
    <pre>
      ReferenceError: styled is not defined
    </pre>
  </div>
</Wrapper>
`;

exports[`basic rendering case snapshot: inline code 1`] = `
<Wrapper
  code="
      (
        <div>
          <span>Hello, world</span>
        </div>
      )
    "
  inline={true}
>
  <div>
    <YozoraCodeRendererJsx
      code="
      (
        <div>
          <span>Hello, world</span>
        </div>
      )
    "
      inline={true}
      onError={[Function]}
      scope={Object {}}
    >
      <ErrorBoundary>
        <div>
          <span>
            Hello, world
          </span>
        </div>
      </ErrorBoundary>
    </YozoraCodeRendererJsx>
    <pre />
  </div>
</Wrapper>
`;
